### PR TITLE
Add string config management for external phone

### PR DESCRIPTION
**Pull request by SebastianMB-IT**
- Add a new section inside Settings where a configuration string can be created for the external phone integrations
- Create a new component that includes phone-island and configure it globally, see more about it on https://github.com/nethesis/phone-island
    - The phone-island is configured using the login authentication token, it can be configured with a specific token too, this one is required and used inside the external phone configuration string
    - The phone-island is currently connecting to the server's janus instance and to the nethcti-servers ws

### CODE REVIEW

**Comment by SebastianMB-IT** [comment]
Some screens:
Screenshot from 2022-11-24 11-32-18
Screenshot from 2022-11-24 11-32-30
Screenshot from 2022-11-24 11-32-43

File: components/settings/Integrations.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            </div>
+            <div>
+              <h4
+                id='phone-configuration-heading'
+                className='text-sm font-medium leading-6 text-gray-900 dark:text-gray-100'
+              >
+                External phone configuration
+              </h4>
+              <p className='mt-1 text-sm text-gray-500 dark:text-gray-400'>
+                Phone configuration can be imported into other applications until integrations are
```
Comment (andre8244) [inline components/settings/Integrations.tsx:83]:
```suggestion
                Anytime the user starts or receives a phone call, a floating phone widget is shown on CTI user interface. This widget can be imported and used into other web applications. Click "Get phone widget configuration" to get the configuration string needed to import the phone widget into your web application.
```

File: components/settings/Integrations.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                ready to work with this one. Creating a new configuration will invalidate the
+                configuration created before. A new configuration string will be shown down here,
+                and you will need to copy and paste it into the new application to configure the
+                phone.
+              </p>
+              <p className='mt-6 flex items-center gap-2'>
+                <Button onClick={newConfig}>Create              {
```
Comment (andre8244) [inline components/settings/Integrations.tsx:90]:
```suggestion
                <Button onClick={newConfig}>Get phone widget configuration              {
```

File: components/settings/Integrations.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            <div>
+              <h4
+                id='phone-configuration-heading'
+                className='text-sm font-medium leading-6 text-gray-900 dark:text-gray-100'
+              >
+                External phone configuration
+              </h4>
+              <p className='mt-1 text-sm text-gray-500 dark:text-gray-400'>
+                Phone configuration can be imported into other applications until integrations are
+                ready to work with this one. Creating a new configuration will invalidate the
```
Comment (andre8244) [inline components/settings/Integrations.tsx:84]: If the user clicks "Create" button the current configuration is immediately invalidated? Maybe it's better to show a confirmation modal with this description: "If you generate a new phone widget configuration, any phone widget in your web applications must be reconfigured with the new string"

Comment (SebastianMB-IT) [reply components/settings/Integrations.tsx:84]: I disabled the creation button instead.

**Comment by SebastianMB-IT** [comment]
After changes:
Screenshot from 2022-11-24 15-41-29

**Comment by SebastianMB-IT** [comment]
The first time:
Screenshot from 2022-11-24 16-00-06

**Review by edospadoni — APPROVED** [review]

**Review by andre8244 — APPROVED** [review]

**Pull request merged**